spec/tbd_tests_spec.rb

```diff
@@ -2637,6 +4196,105 @@
+    expect(vals.empty?).to be(false)
+    expect(holds[:fenestration]).to be(true)
+    expect(vals[:sill]).to be_within(0.001).of(0.371)
+    expect(vals[:sillconcave]).to be_within(0.001).of(0.372)
+    expect(vals[:sillconvex]).to  be_within(0.001).of(0.373)
+
+    expect(psi.set.has_key?("partial sills")).to be(false)
+    partial_sills =
+    {
+      id:            "partial sills",
+      fenestration:  0.391,
+      head:          0.381,
+      headconcave:   0.382,
+      headconvex:    0.383,
+      sill:          0.371,
+      sillconcave:   0.372,
+      # sillconvex:    0.373,                      # dropping the convex variant
+      jamb:          0.361,
+      jambconcave:   0.362,
+      jambconvex:    0.363,
+      rimjoist:      0.001,
+      parapet:       0.002,
+      corner:        0.003,
+      balcony:       0.004,
+      party:         0.005,
+      grade:         0.006
+    }
+    psi.append(partial_sills)
+    expect(psi.set.has_key?("partial sills")).to be(true)
+    expect(psi.complete?("partial sills")).to be(true)   # can be a building set
+    holds, vals = psi.shorthands("partial sills")
+    expect(holds.empty?).to be(false)
+    expect(vals.empty?).to be(false)
+    expect(holds[:sillconvex]).to be(false)                # absent from PSI set
+    expect(vals[:sill]).to        be_within(0.001).of(0.371)
+    expect(vals[:sillconcave]).to be_within(0.001).of(0.372)
+    expect(vals[:sillconvex]).to  be_within(0.001).of(0.371)    # inherits :sill
+
+    expect(psi.set.has_key?("no sills")).to be(false)
+    no_sills =
+    {
+      id:            "no sills",
+      fenestration:  0.391,
+      head:          0.381,
+      headconcave:   0.382,
+      headconvex:    0.383,
+      # sill:          0.371,                     # dropping the concave variant
+      # sillconcave:   0.372,                     # dropping the concave variant
+      # sillconvex:    0.373,                      # dropping the convex variant
+      jamb:          0.361,
+      jambconcave:   0.362,
+      jambconvex:    0.363,
+      rimjoist:      0.001,
+      parapet:       0.002,
+      corner:        0.003,
+      balcony:       0.004,
+      party:         0.005,
+      grade:         0.006
+    }
+    psi.append(no_sills)
+    expect(psi.set.has_key?("no sills")).to be(true)
+    expect(psi.complete?("no sills")).to be(true)        # can be a building set
+    holds, vals = psi.shorthands("no sills")
+    expect(holds.empty?).to be(false)
+    expect(vals.empty?).to be(false)
+    expect(holds[:sill]).to be(false)                      # absent from PSI set
+    expect(holds[:sillconcave]).to be(false)               # absent from PSI set
+    expect(holds[:sillconvex]).to be(false)                # absent from PSI set
+    expect(vals[:sill]).to        be_within(0.001).of(0.391)
+    expect(vals[:sillconcave]).to be_within(0.001).of(0.391)
+    expect(vals[:sillconvex]).to  be_within(0.001).of(0.391)     # :fenestration
```

Comment:
The _complete?_ method always returns "true", whether PSI sets are holding all, some or no _sill_ items. The _holds_ hash correctly reports what the PSI set carries, while _vals_ returns inherited PSI values.